### PR TITLE
fix(layout): fix elevation for posts

### LINKED ISSUE #54: 文章卡片的海拔(elevation)可以高一点
文章卡片看起来好像没有设置elevation，即纸片的阴影
参考：
[https://material.google.com/material-design/elevation-shadows.html#elevation-shadows-elevation-android](url)

### PR DESCRIPTION
**Pull request by simonsmh**
**Contributing rules**

- Fork the repo and create your branch from `canary`. Then be sure to put the `canary` branch as the target for your pull request.
- Please be sure to follow the [contributing guidelines](https://github.com/viosey/hexo-theme-material/blob/master/CONTRIBUTING.md)

**What kind of change does this PR introduce?** (check one with "x")

- [x] Bug fix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

image
[Fix #54](https://github.com/viosey/hexo-theme-material/issues/54)
[Reference: cards-section](https://getmdl.io/components/index.html#cards-section)
to review

### CODE REVIEW

**Comment by simonsmh** [comment]
烦透了！！

**Comment by cubesky** [comment]
@simonsmh 我第一次用的时候跟他奋斗了2个小时

**Comment by simonsmh** [comment]
done done done

**Comment by cubesky** [comment]
@simonsmh 恭喜，第一次遇到这个确实挺难过的

**Comment by viosey** [comment]
I haven't decided whether to change this elevation.
It's a preference, so at least we should add a configuration.

**Comment by simonsmh** [comment]
@viosey  
Please read https://material.google.com/material-design/elevation-shadows.html#elevation-shadows-elevation-android  

> Don't.
> Without a shadow, nothing indicates that the floating action button is separate from the background surfaces.

如果这算是material design，那么就应当采用card应有的形式。Google mdl的template只是个sample，项目可以自己修改，不用完全照搬啊。
此外，post有相关的elevation设置，与主页的设置不符，造成视觉不适，以及logo区域圆角矩形CSS与card CSS设置亦不相同，造成视觉冲突，此pr也一同修复。

**Comment by viosey** [comment]
@simonsmh 这套 scheme 叫做 `paradox` 的原意就是表明并非纯粹的 Material Design，而 nexus scheme 才遵守 Material Design。
如果要修改 elevation 的话需要以自定义功能的形式，不能写死。

**Comment by simonsmh** [comment]
好的，不过大概明天晚上才能改了
今天还得写作业（唉

**Comment by simonsmh** [comment]
done

**Comment by viosey** [comment]
呃，可能我没有说清楚，就是配置的话不要使用开关的形式，而是自定义高度，以数值为参数

**Comment by simonsmh** [comment]
viosey是做产品经理的么

Viosey <notifications@github.com>于2017年1月20日周五 16:02写道：

**Comment by Halyul** [comment]
分明是一只大学汪

**Comment by simonsmh** [comment]
是这种样子的嘛？

**Comment by viosey** [comment]
差不多了，我再修改一下

File: source/css/style.css [hunk 1]
```diff
@@ -552,7 +552,7 @@ a {
     min-height: 400px;
     flex-direction: column;
     align-items: stretch;
-  border-radius: 5px;
+  border-radius: 2px;
```
Comment (pidupuis) [inline source/css/style.css]: - Please fix tab (use 4 tabs)

File: layout/_partial/Isolation-post_entry.ejs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<div class="post_entry-module mdl-card mdl-cell mdl-cell--12-col fade out">
+<div class="post_entry-module mdl-card <% if(theme.card_elevation) { %>mdl-shadow--<%= theme.card_elevation %>dp<% } %> mdl-cell mdl-cell--12-col fade out">
     <!-- Post_entry Header -->
     <% if(post.thumbnail == undefined) { %>
```
Comment (pidupuis) [inline layout/_partial/Isolation-post_entry.ejs]: - Please use `<% if(!post.thumbnail) { %>`

File: source/css/style.css [hunk 3]
```diff
@@ -1287,7 +1287,7 @@ a {
 #scheme-Isolation .post_entry-module {
   margin-bottom: 24px;
   width: 100%;
-    border-radius: 5px;
+    border-radius: 2px;
```
Comment (pidupuis) [inline source/css/style.css]: - Please fix tab (use 4 tabs)

File: source/css/style.css [hunk 4]
```diff
@@ -1812,7 +1812,7 @@ a {
 
 #scheme-Isolation .material-post .mdl-card{
   width: 100%;
-  border-radius: 5px;
+  border-radius: 2px;
```
Comment (pidupuis) [inline source/css/style.css]: - Please fix tab (use 4 tabs)

File: layout/_partial/Paradox-post_entry.ejs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<div class="post_entry-module mdl-card mdl-cell mdl-cell--12-col fade out">
+<div class="post_entry-module mdl-card <% if(theme.card_elevation) { %>mdl-shadow--<%= theme.card_elevation %>dp<% } %> mdl-cell mdl-cell--12-col fade out">
```
Comment (pidupuis) [inline layout/_partial/Paradox-post_entry.ejs]: I would use `<% if(theme.card_elevation) { %><%= 'mdl-shadow--' + theme.card_elevation + 'dp' %><% } %>` for more readability

File: layout/_partial/blog_info.ejs [hunk 6]
```diff
@@ -1,5 +1,5 @@
 <!-- Blog info -->
-<div class="mdl-card something-else mdl-cell mdl-cell--8-col mdl-cell--4-col-desktop index-top-block">
+<div class="mdl-card <% if(theme.card_elevation) { %>mdl-shadow--<%= theme.card_elevation %>dp<% } %> something-else mdl-cell mdl-cell--8-col mdl-cell--4-col-desktop index-top-block">
```
Comment (pidupuis) [inline layout/_partial/blog_info.ejs]: I would use `<% if(theme.card_elevation) { %><%= 'mdl-shadow--' + theme.card_elevation + 'dp' %><% } %>` for more readability

File: layout/_partial/daily_pic.ejs [hunk 7]
```diff
@@ -1,5 +1,5 @@
 <!-- Daily pic -->
-<div class="mdl-card daily-pic mdl-cell mdl-cell--8-col index-top-block">
+<div class="mdl-card <% if(theme.card_elevation) { %>mdl-shadow--<%= theme.card_elevation %>dp<% } %> daily-pic mdl-cell mdl-cell--8-col index-top-block">
```
Comment (pidupuis) [inline layout/_partial/daily_pic.ejs]: I would use `<% if(theme.card_elevation) { %><%= 'mdl-shadow--' + theme.card_elevation + 'dp' %><% } %>` for more readability

File: layout/post.ejs [hunk 8]
```diff
@@ -16,7 +16,7 @@
     <!-- Hidden Page -->
     <div class="material-post_container">
         <div class="material-post mdl-grid">
-            <div class="mdl-card mdl-shadow--4dp mdl-cell mdl-cell--12-col">
+            <div class="mdl-card <% if(theme.card_elevation) { %>mdl-shadow--<%= theme.card_elevation %>dp<% } %> mdl-cell mdl-cell--12-col">
```
Comment (pidupuis) [inline layout/post.ejs]: I would use `<% if(theme.card_elevation) { %><%= 'mdl-shadow--' + theme.card_elevation + 'dp' %><% } %>` for more readability

**Comment by simonsmh** [comment]
Review please.

Pierre Dupuis <notifications@github.com>于2017年1月20日周五 20:13写道：

**Comment by pidupuis** [comment]
Don't worry about gitmagic for this time, your commit message is good. It was bad config from our part sorry

**Pull request merged**